destination/format/csv.go

```diff
@@ -21,26 +21,117 @@ import (
 	"github.com/go-errors/errors"
+	"github.com/hamba/avro"
 	"golang.org/x/exp/maps"
 )
 
+const YYYYMMDD = "2006-01-02"
+
+// Map between snowflake retrieved types and connector defined types
+// TODO: just create the table with the types on the left to make this simpler.
+var SnowflakeTypeMapping = map[string]string{
+	SnowflakeFixed:        SnowflakeInteger,
+	SnowflakeReal:         SnowflakeFloat,
+	SnowflakeText:         SnowflakeVarchar,
+	SnowflakeBinary:       SnowflakeBinary,
+	SnowflakeBoolean:      SnowflakeBoolean,
+	SnowflakeDate:         SnowflakeDate,
+	SnowflakeTimestampNTZ: SnowflakeTimestampNTZ,
+	SnowflakeTimestampLTZ: SnowflakeTimestampLTZ,
+	SnowflakeTimestampTZ:  SnowflakeTimestampTZ,
+	SnowflakeVariant:      SnowflakeVariant,
+	SnowflakeObject:       SnowflakeObject,
+	SnowflakeArray:        SnowflakeArray,
+	SnowflakeVector:       SnowflakeVector,
+}
+
+const (
+	// numeric types
+	SnowflakeInteger = "INTEGER"
+	SnowflakeFloat   = "FLOAT"
+	SnowflakeFixed   = "FIXED"
+	SnowflakeReal    = "REAL"
+
+	// string & binary types
+	SnowflakeText    = "TEXT"
+	SnowflakeBinary  = "BINARY"
+	SnowflakeVarchar = "VARCHAR"
+
+	// logical data types
+	SnowflakeBoolean = "BOOLEAN"
+
+	// date & time types
+	SnowflakeTimestampLTZ = "TIMESTAMP_LTZ"
+	SnowflakeTimestampNTZ = "TIMESTAMP_NTZ"
+	SnowflakeTimestampTZ  = "TIMESTAMP_TZ"
+	SnowflakeTime         = "TIME"
+	SnowflakeDate         = "DATE"
+
+	// semi-structured data types
+	SnowflakeVariant = "VARIANT"
+	SnowflakeObject  = "OBJECT"
+	SnowflakeArray   = "ARRAY"
+
+	// geospatial data types
+	SnowflakeGeography = "GEOGRAPHY"
+	SnowflakeGeometry  = "GEOMETRY"
+
+	// vector data types
+	SnowflakeVector = "VECTOR"
+)
+
+// Map from Avro Types to Snowflake Types
+// TODO: be more precise about numeric types.
+var AvroToSnowflakeType = map[string]string{
+	AvroBoolean:         SnowflakeBoolean,
+	AvroInt:             SnowflakeInteger,
+	AvroLong:            SnowflakeInteger,
+	AvroFloat:           SnowflakeFloat,
+	AvroDouble:          SnowflakeFloat,
+	AvroDecimal:         SnowflakeFloat,
+	AvroBytes:           SnowflakeVarchar,
+	AvroString:          SnowflakeVarchar,
+	AvroUUID:            SnowflakeVarchar,
+	AvroDate:            SnowflakeDate,
+	AvroTimeMillis:      SnowflakeTimestampLTZ,
+	AvroTimeMicros:      SnowflakeTimestampLTZ,
+	AvroTimestampMillis: SnowflakeTimestampLTZ,
+	AvroTimestampMicros: SnowflakeTimestampLTZ,
+
+	AvroRecord: SnowflakeObject,
+	AvroArray:  SnowflakeArray,
+	AvroMap:    SnowflakeObject,
+}
+
 const (
-	snowflakeTimeStamp = "TIMESTAMP_LTZ"
-	snowflakeVarchar   = "VARCHAR"
-	snowflakeVariant   = "VARIANT"
-	snowflakeInteger   = "INTEGER"
-	snowflakeBoolean   = "BOOLEAN"
-	snowflakeFloat     = "FLOAT"
+	AvroBoolean         = "boolean"
+	AvroInt             = "int"
+	AvroLong            = "long"
+	AvroFloat           = "float"
+	AvroDouble          = "double"
+	AvroBytes           = "bytes"
+	AvroString          = "string"
+	AvroDecimal         = "decimal"
```

Comment:
there isn't an explicit decimal type, but rather `fixed` with logical subtype decimal

Reply:
yeah, but isn't this the same with `date`?

Reply:
kind of but not the same, dates/timestamps are represented as long, you will need to look in the logical subtype to determine what is it.  These need to be handle separately like this, won't be a straight forward map resolver.

```
p := field.Type().(*avro.PrimitiveSchema)
switch t := p.Logical().(type) {
case avro.Date:
 ....
case ...

}
```